config/config_test.go

```diff
@@ -0,0 +1,30 @@
+package config
+import (
+	. "github.com/smartystreets/goconvey/convey"
+	"os"
+	"testing"
+)
+func TestGetGoFigureErrors(t *testing.T) {
+	Convey("When I try to get the config via the Get method", t, func(){
+		cfg, err := Get()
+		Convey("Then I am given an error when calling go figure and it is handled correctly", func() {
+			So(cfg, ShouldBeNil)
+			So(err, ShouldNotBeNil)
+		})
+	})
+}
+func TestGetNoErrors(t *testing.T) {
+	_ = os.Setenv("BIND_ADDR", "bind_addr")
```

Comment:
These unit tests are not exhaustive. As far as I can see, this test will not show an issue if the code fails to complain that a given mandatory element is missing because it is already complaining that another element is missing. We need a test per mandatory element, don't we?
Happy to punt this additional test code into a subsequent ticket.

Reply:
I agree but at the same time, we usually only unit test exported methods (correct me if i'm wrong here!) So the line that is covered is if this local method returns an error, is it handled correctly....

We can unit test the method if we want, but I don't believe we do usually.

Reply:
I suppose what I am talking about is unit testing that the struct is correctly written - i.e. unit testing the behaviour of the app, asserting that its behaviour around mandatory config is as-designed.
If not this test to assert that errors in bad config are handled, then where?

Reply:
discussed in a slack chat to avoid heavily weighing down this PR with comments.

Outcome: While working jointly between Faisal and I, it seems a unit test has been wrongly written in it's function name / Conveys to suggest it is covering calls to GoFigure. Whereas, it is in fact covering the mandatory validation private method call instead. Faisal will push an update so sort this out to avoid confusion. Commenting here so that it is documented when looking back at PR.